src/event-engine/states/receive_failed.ts

```diff
@@ -0,0 +1,45 @@
+import { State } from '../core/state';
+import { Cursor } from '../../models/Cursor';
+import { Effects } from '../effects';
+import { Events, reconnect, restore, subscriptionChange, unsubscribeAll } from '../events';
+import { PubNubError } from '../../core/components/endpoint';
+import { HandshakingState } from './handshaking';
+import { UnsubscribedState } from './unsubscribed';
+
+export type ReceiveFailedStateContext = {
+  channels: string[];
+  groups: string[];
+  cursor: Cursor;
+
+  reason: PubNubError;
+};
+
+export const ReceiveFailedState = new State<ReceiveFailedStateContext, Events, Effects>('RECEIVE_FAILED');
+
+ReceiveFailedState.on(reconnect.type, (context, event) =>
+  HandshakingState.with({
+    channels: context.channels,
+    groups: context.groups,
+    cursor: {
+      timetoken: event.payload?.timetoken ?? '0',
+      region: event.payload?.region ?? 0,
+    },
+  }),
+);
+
+ReceiveFailedState.on(subscriptionChange.type, (_, event) =>
+  HandshakingState.with({
+    channels: event.payload.channels,
+    groups: event.payload.groups,
+  }),
+);
+
+ReceiveFailedState.on(restore.type, (_, event) =>
+  HandshakingState.with({
+    channels: event.payload.channels,
+    groups: event.payload.groups,
+    cursor: { timetoken: event.payload.timetoken, region: event.payload?.region ?? 0 },
```

Comment:
In other states, we used here region from the context instead of `0`.

Reply:
Updating, 

The reason was:
here, It's getting into Handshaking.
So,
Once it get successfull response for handshake, It will anyway use server provided region (along with user provided tt) with the thought that it's recommended to use server provided region value

Reply:
Ok. Got the point. Let it be `0` :)

Reply:
unified across the state: and when values are not provided by user and also state context doesn't have it.. Then only it will default to 0 which indicates falsy value while handshaking